bbc/simorgh
# WSTEAM1-240: Fix a11y bug by changing Image component

Resolves JIRA [WSTEAM1-240](https://jira.dev.bbc.co.uk/browse/WSTEAM1-240) - subtask https://jira.dev.bbc.co.uk/browse/WSTEAM1-791

Bug fix for https://github.com/bbc/simorgh/pull/11273

Overall changes
======
Fix an A11y bug that occurs when the browser is zoomed in up to 200% and/or there is a large amount of text

Bug Description
======
When the browser is zoomed in and/or there is a large amount of text, the header can exceed 440px in height. When this happens, the image becomes wider because our Image component has a fixed image ratio.

In my initial solution, I used a image-mask to apply the gradient. This scales with the image and it does not provide adequate background to the text if the image becomes to wide.

Public Service get around this issue by changing the aspect ratio of the image. However our Image component does not allow for this currently.

So I have updated our Image component to allow us to not have a fixed aspect ratio. This matches PS implementation and designs.

Old PR (with bug - [storybook link](https://wsteam1-240-header-image--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-livepageheader--title-and-description-with-live-label-and-image&viewMode=story)):

<img width="1105" alt="Screenshot 2024-01-29 at 10 57 20" src="https://github.com/bbc/simorgh/assets/98817636/07e58eeb-960d-40fc-a5ee-e53345a74102">

This PR (bug resolved)

<img width="1078" alt="Screenshot 2024-01-29 at 15 59 17" src="https://github.com/bbc/simorgh/assets/98817636/01d241a9-85c2-438c-b6cd-316186ac6265">

Code changes
======
(from https://github.com/bbc/simorgh/pull/11273)
- Updates Image Component to not have a fixed Aspect Ratio if no width & height or aspect ratio is provided. 
- Update other components affected by this and snapshots.
- Removed aspectRatio prop from MaskedImage
- Add extra long text storybook version for UX review.

Testing
======
Testing Locally

- pull down this branch
- cd into ws-nextjs-app and use yarn dev to run the application locally
- Visit http://localhost:7081/pidgin/live/c0mrezv7rd0t to see a header image (LTR)
- Visit http://localhost:7081/urdu/live/cd0jyl9r0lzt to see a header image (RTL)
- Visit http://localhost:7081/pidgin/live/c7p765ynk9qt to see a text-only header

Storybook permalink: https://wsteam1-240-custom-image--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-livepageheader--title-and-description-with-live-label-and-image-extra-long-text&viewMode=story

Helpful Links
======
[UX designs (header image)](https://www.figma.com/file/D9w5JGLOhznDlz5tHu3rH0?node-id=5:63&locale=en&type=design)

[Screen reader UX Figma (header image)](https://www.figma.com/file/D9w5JGLOhznDlz5tHu3rH0/Live-mvp---Live-Topic-Header---handoff?type=design&node-id=104-10&mode=design&t=okUjISN0SwzmMfXw-0)

[Live Page Screen Reader UX](https://paper.dropbox.com/doc/WS-Live-page-Screen-reader-UX--CIFvUftYG2EdtFiNsaBzR9ovAg-8qm1UHDGVMhv5Qj9Q2mbi)

[Live Page A11y A/Cs](https://paper.dropbox.com/doc/Live-page-Accessibility-Acceptance-Criteria--CIHCd~3oDcJAv_Zt_ucLcbbIAg-KZQODmAf9tvodUlxsAVnA)

[UX designs (text only header)](https://www.figma.com/file/ozHsWG5R9tETw6lBOU740V/Live-mvp-header-(text-only)---handover?type=design&node-id=1-2&mode=design&t=xxJSN2MlwLiK80Ei-0)

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
